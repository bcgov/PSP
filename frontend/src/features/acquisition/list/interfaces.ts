export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
}
