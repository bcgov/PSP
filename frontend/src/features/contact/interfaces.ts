export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}
