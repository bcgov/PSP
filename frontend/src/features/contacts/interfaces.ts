export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}

export enum ContactTypes {
  ORGANIZATION = 'O',
  INDIVIDUAL = 'P',
}
