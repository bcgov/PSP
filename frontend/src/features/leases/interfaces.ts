export interface ILeaseFilter {
  pinOrPid: string;
  lFileNo: string;
  searchBy: string;
  tenantName: string;
  programs: string[];
}
