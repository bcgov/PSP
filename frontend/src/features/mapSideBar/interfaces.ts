import { AddressTypes, PropertyTypes } from 'constants/index';
import { IAddress } from 'interfaces';

export const defaultPropertyValues = {
  address: {
    addressTypeId: AddressTypes.Physical,
    streetAddress1: '',
    municipality: '',
    postal: '',
    province: '',
    provinceId: 1,
    region: '',
    district: '',
  } as IAddress,
  pid: '',
  propertyTypeId: PropertyTypes.Land,
  classification: '',
  region: '',
  district: '',
  zoning: '',
  landArea: '',
  landLegalDescription: '',

  // TODO: Theses properties probably should be replaced or updated.
  titleNumber: '',
  purpose: '',
  ruralArea: '',
  ocpDesignation: '',
};
