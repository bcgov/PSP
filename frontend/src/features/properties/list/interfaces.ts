import { IAddress, IBuilding, IEvaluation, IFiscal } from 'interfaces';

/**
 * IProperty interface represents the model used for searching properties.
 */
export interface IProperty {
  id: number;
  propertyType: string;
  propertyTypeId: number;
  pid: string;
  pin?: number;
  name?: string;
  classificationId: number;
  classification: string;
  description: string;
  latitude: number;
  longitude: number;
  isSensitive: boolean;
  agencyId: number;
  agency: string;
  agencyCode: string;
  subAgency?: string;
  subAgencyCode?: string;

  addressId: number;
  address: string;
  administrativeArea: string;
  province: string;
  postal: string;
  city: string;

  // Financial Values
  market: number;
  marketFiscalYear?: number;
  netBook: number;
  netBookFiscalYear?: number;

  assessedLand?: number;
  assessedLandDate?: Date | string;
  assessedBuilding?: number;
  assessedBuildingDate?: Date | string;

  // Parcel Properties
  landArea: number;
  landLegalDescription: string;
  zoning?: string;
  zoningPotential?: string;

  // Building Properties
  parcelId?: number;
  constructionTypeId?: number;
  constructionType?: string;
  predominateUseId?: number;
  predominateUse?: string;
  occupantTypeId?: number;
  occupantType?: string;
  floorCount?: number;
  tenancy?: string;
  occupantName?: string;
  leaseExpiry?: Date | string;
  transferLeaseOnSale?: boolean;
  rentableArea?: number;
}

/**
 * IPropertyQueryParams interface, provides a model for querying the API for properties.
 */
export interface IPropertyQueryParams {
  page: number;
  quantity: number;
  pid?: string;
  address?: string;
  name?: string;
  administrativeArea?: string;
  classificationId?: number;
  agencies?: number | number[];
  minLandArea?: number;
  maxLandArea?: number;
  minLotArea?: number;
  maxLotArea?: number;
  all?: boolean;
  parcelId?: number;
  propertyType?: string;
  bareLandOnly?: boolean;
  maxNetBookValue?: number | string;
  maxAssessedValue?: number | string;
  maxMarketValue?: number | string;
}

export interface IParentParcel {
  pid: string;
  pin: number;
  id: number;
}

/**
 * IProperty interface represents the model used for displaying properties.
 */
export interface IFlatProperty {
  id: number;
  projectPropertyId?: number;
  propertyTypeId: number;
  propertyType: string;
  pid: string;
  pin?: number;
  classificationId: number;
  classification: string;
  name: string;
  description: string;
  projectNumbers?: string[];
  latitude: number;
  longitude: number;
  isSensitive: boolean;
  agencyId: number;
  agency: string;
  agencyCode: string;
  subAgency?: string;
  subAgencyCode?: string;

  addressId: number;
  address: string;
  administrativeArea: string;
  province: string;
  postal: string;

  // Financial Values
  market: number | '';
  marketFiscalYear?: number;
  marketRowVersion?: number;
  netBook: number | '';
  netBookFiscalYear?: number;
  netBookRowVersion?: number;

  assessedLand?: number | '';
  assessedLandDate?: Date | string;
  assessedLandFirm?: string;
  assessedLandRowVersion?: number;
  assessedBuilding?: number | '';
  assessedBuildingDate?: Date | string;
  assessedBuildingFirm?: string;
  assessedBuildingRowVersion?: number;

  // Parcel Properties
  landArea: number;
  landLegalDescription: string;
  zoning?: string;
  zoningPotential?: string;
  parcels?: IParentParcel[];

  // Building Properties
  parcelId?: number;
  constructionTypeId?: number;
  constructionType?: string;
  predominateUseId?: number;
  predominateUse?: string;
  occupantTypeId?: number;
  occupantType?: string;
  floorCount?: number;
  tenancy?: string;
  occupantName?: string;
  leaseExpiry?: Date | string;
  transferLeaseOnSale?: boolean;
  rentableArea?: number;
  rowVersion?: number;
}

export interface IApiProperty {
  id: number;
  parcelId?: number;
  buildingId?: number;
  propertyTypeId: number;
  pid?: string;
  pin?: number | '';
  projectNumbers: string[];
  latitude: number;
  longitude: number;
  classification?: string;
  classificationId: number;
  name: string;
  description: string;
  address?: IAddress;
  landArea: number;
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency?: string;
  subAgency?: string;
  agencyId: number;
  isSensitive: boolean;
  buildings: IBuilding[];
  evaluations: IEvaluation[];
  fiscals: IFiscal[];
  parcels?: IParentParcel[];
  rowVersion?: number;
}
