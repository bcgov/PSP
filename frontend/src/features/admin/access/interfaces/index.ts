import { AccessRequestStatus } from 'constants/accessStatus';

export interface IAccessRequestModel {
  id: number;
  userId: number;
  businessIdentifierValue: string;
  firstName: string;
  surname: string;
  email: string;
  position: string;
  status: AccessRequestStatus;
  organization: string;
  role: string;
  note: string;
}
