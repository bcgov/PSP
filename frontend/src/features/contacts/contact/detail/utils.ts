import { IContactAddress } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
