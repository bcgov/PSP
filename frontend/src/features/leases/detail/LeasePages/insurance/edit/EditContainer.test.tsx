import userEvent from '@testing-library/user-event';
import { Formik } from 'formik';
import { createMemoryHistory } from 'history';
import { IInsurance, TypeCodeUtils } from 'interfaces';
import { noop } from 'lodash';
import { ILookupCode } from 'store/slices/lookupCodes';
import { act, render, RenderOptions, RenderResult } from 'utils/test-utils';

import InsuranceEditContainer, { InsuranceEditContainerProps } from './EditContainer';

const mockInsuranceTypeHome: ILookupCode = {
  id: 'HOME',
  name: 'Home Insurance',
  type: 'PimsInsuranceType',
  isDisabled: false,
  displayOrder: 1,
};
const mockInsuranceTypeCar: ILookupCode = {
  id: 'CAR',
  name: 'Car insurance',
  type: 'PimsInsuranceType',
  isDisabled: false,
  displayOrder: 2,
};

const mockInsuranceHome: IInsurance = {
  id: 123459,
  insuranceType: TypeCodeUtils.createFromLookup(mockInsuranceTypeHome),
  otherInsuranceType: '',
  coverageDescription: '',
  coverageLimit: 777,
  expiryDate: '2022-01-01',
  isInsuranceInPlace: true,
  rowVersion: 0,
};

const defaultProps: InsuranceEditContainerProps = {
  leaseId: 1,
  insuranceList: [mockInsuranceHome],
  insuranceTypes: [mockInsuranceTypeHome, mockInsuranceTypeCar],
  onCancel: () => {},
  onSuccess: () => {},
};

const history = createMemoryHistory();

describe('Lease Insurance', () => {
  const setup = (
    renderOptions: RenderOptions & InsuranceEditContainerProps = {
      leaseId: defaultProps.leaseId,
      insuranceList: defaultProps.insuranceList,
      insuranceTypes: defaultProps.insuranceTypes,
      onCancel: () => defaultProps.onCancel,
      onSuccess: () => defaultProps.onSuccess,
    },
  ): RenderResult => {
    // render component under test
    const result = render(
      <Formik onSubmit={noop} initialValues={{}}>
        <InsuranceEditContainer
          insuranceList={renderOptions.insuranceList}
          insuranceTypes={renderOptions.insuranceTypes}
          leaseId={renderOptions.leaseId}
          onCancel={renderOptions.onCancel}
          onSuccess={renderOptions.onSuccess}
        />
      </Formik>,
      {
        ...renderOptions,
        history,
      },
    );
    return result;
  };

  it('renders as expected', () => {
    const result = setup();
    expect(result.asFragment()).toMatchSnapshot();
  });

  it('Shows correct comboboxes', () => {
    const result = setup({
      ...defaultProps,
      insuranceList: [mockInsuranceHome],
      insuranceTypes: [mockInsuranceTypeHome, mockInsuranceTypeCar],
    });
    const checkboxes = result.getAllByTestId('insurance-checkbox');
    expect(checkboxes.length).toBe(2);
    expect((checkboxes[0] as HTMLInputElement).value).toBe(mockInsuranceTypeHome.id);
    expect((checkboxes[1] as HTMLInputElement).value).toBe(mockInsuranceTypeCar.id);

    expect(checkboxes[0] as HTMLInputElement).toBeChecked();
    expect(checkboxes[1] as HTMLInputElement).not.toBeChecked();
  });

  it('Updates form lists when clicked', async () => {
    const testInsuranceCar: IInsurance = {
      ...mockInsuranceHome,
      coverageLimit: 888,
      coverageDescription: 'test description for a test insurance car',
    };
    testInsuranceCar.insuranceType = TypeCodeUtils.createFromLookup(mockInsuranceTypeCar);

    const result = setup({
      ...defaultProps,
      insuranceList: [mockInsuranceHome, testInsuranceCar],
      insuranceTypes: [mockInsuranceTypeHome, mockInsuranceTypeCar],
    });

    // Verify initial state
    const forms = result.getAllByTestId('insurance-form');
    expect(forms.length).toBe(2);

    const formLimits = result.getAllByTestId('insurance-form-limit');
    expect(formLimits[0] as HTMLInputElement).toHaveValue(mockInsuranceHome.coverageLimit);
    expect(formLimits[1] as HTMLInputElement).toHaveValue(testInsuranceCar.coverageLimit);

    const formDescriptions = result.getAllByTestId('insurance-form-description');
    expect(formDescriptions[0] as HTMLInputElement).toHaveValue(
      mockInsuranceHome.coverageDescription,
    );
    expect(formDescriptions[1] as HTMLInputElement).toHaveValue(
      testInsuranceCar.coverageDescription,
    );

    // Update view
    const checkboxes = result.getAllByTestId('insurance-checkbox');

    await act(async () => userEvent.click(checkboxes[0]));

    // Verify update
    const updatedForms = result.getAllByTestId('insurance-form');
    expect(updatedForms.length).toBe(1);

    const formTitle = result.getByTestId('insurance-form-title');
    expect(formTitle.textContent).toBe(mockInsuranceTypeCar.name);
  });
});
