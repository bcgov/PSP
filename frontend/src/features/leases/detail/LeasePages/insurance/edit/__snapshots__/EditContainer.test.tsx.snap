// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lease Insurance renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="Toastify"
  />
  <h2>
    Required coverage
  </h2>
  <div>
    Select the coverage types that are required for this lease or license.
  </div>
  .c0 {
  border-radius: 1rem;
  width: 100%;
  padding-top: 1rem;
  padding-left: 2.5rem;
  padding-bottom: 1rem;
}

<div
    class="c0"
  >
    <div
      class="form-group"
    >
      <div
        class="form-check"
      >
        <input
          checked=""
          class="form-check-input"
          data-testid="insurance-checkbox"
          id="insurance-0"
          name="checkedTypes"
          type="checkbox"
          value="HOME"
        />
        <label
          class="form-check-label"
          for="insurance-0"
        >
          Home Insurance
        </label>
      </div>
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          data-testid="insurance-checkbox"
          id="insurance-1"
          name="checkedTypes"
          type="checkbox"
          value="CAR"
        />
        <label
          class="form-check-label"
          for="insurance-1"
        >
          Car insurance
        </label>
      </div>
    </div>
  </div>
  <h2>
    Coverage details
  </h2>
  .c4.c4.form-control.is-valid {
  background-image: none;
}

.c4.c4.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 0.8rem;
}

.c3 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  border-radius: 1rem;
  width: 100%;
  padding: 2.5rem;
}

.c1.form-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0;
}

.c1.form-group .radio-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  row-gap: unset;
  -webkit-column-gap: 0.4rem;
  column-gap: 0.4rem;
}

.c1.form-group .radio-group .form-check {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1.form-group .radio-group .form-check .form-check-input {
  margin-left: 0;
}

.c1.form-group .radio-group .form-check-label {
  margin-left: 1rem;
}

.c1 .form-label {
  margin-bottom: unset;
}

<div>
    <div
      class="c0 mb-3"
      data-testid="insurance-form"
    >
      <h2
        class=""
        data-testid="insurance-form-title"
      >
        Home Insurance
      </h2>
      <div
        class="py-4 row"
      >
        <div
          class="col-md-auto"
        >
          <strong>
            Insurance In Place:
          </strong>
        </div>
        <div
          class="col"
        >
          <div
            class="c1 form-group"
          >
            <div
              class="radio-group"
            >
              <div
                class="c2"
              >
                <div
                  class="form-check"
                >
                  <input
                    checked=""
                    class="form-check-input position-static"
                    id="input-yes"
                    name="insurances.0.isInsuranceInPlaceRadio"
                    type="radio"
                    value="yes"
                  />
                </div>
                <label
                  class="form-check-label form-label"
                  for="input-insurances.0.isInsuranceInPlaceRadio"
                >
                  Yes
                </label>
              </div>
              <div
                class="c2"
              >
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input position-static"
                    id="input-no"
                    name="insurances.0.isInsuranceInPlaceRadio"
                    type="radio"
                    value="no"
                  />
                </div>
                <label
                  class="form-check-label form-label"
                  for="input-insurances.0.isInsuranceInPlaceRadio"
                >
                  No
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <strong>
              Limit ($)
            </strong>
          </div>
          <div
            class="c3 input form-group"
          >
            <input
              class="form-control"
              data-testid="insurance-form-limit"
              id="input-insurances.0.coverageLimit"
              name="insurances.0.coverageLimit"
              type="number"
              value="777"
            />
          </div>
          <div>
            <strong>
              Policy Expiry date:
            </strong>
          </div>
          <div
            class="form-group"
          >
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  autocomplete="off"
                  class="c4 form-control date-picker"
                  id="datepicker-insurances.0.expiryDate"
                  name="insurances.0.expiryDate"
                  placeholder="--/--/----"
                  type="text"
                  value="01/01/2022"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-8"
        >
          <div>
            <strong>
              Description of Coverage
            </strong>
          </div>
          <div
            class="textarea input form-group"
          >
            <textarea
              class="form-control"
              data-testid="insurance-form-description"
              id="input-insurances.0.coverageDescription"
              name="insurances.0.coverageDescription"
              rows="4"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="justify-content-md-end row"
  >
    <div
      class="cancelSave btn-toolbar"
      role="toolbar"
    >
      <div
        class="col"
      >
        <button
          class="mr-5 btn btn-secondary"
          type="button"
        >
          Cancel
        </button>
      </div>
      <div
        class="col"
      >
        <button
          class="mr-5 btn btn-primary"
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
